BeaverName.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Text.RegularExpressions;
+
+namespace NameThatBeaver
+{
+    public class BeaverName
+    {
+        public string Name { get; set; } = string.Empty;
+        public string? Color { get; set; }
+        public long Id { get; private set; }
+
+        public BeaverName(string name, string? color)
+        {
+            Name = name;
+            Color = color;
+            Id = GetHashCode();
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hash = 17;
+                hash = hash * 23 + Name.GetHashCode();
+                if (!string.IsNullOrWhiteSpace(Color))
+                {
+                    hash = hash * 23 + Color.GetHashCode();
+                }
+                return hash;
+            }
+        }
+
+        public override string ToString()
+        {
+            return string.IsNullOrWhiteSpace(Color) ?
+                Name :
+                $"<color=#{Color}>{Name}";
```

Comment:
What do you think about adding `</color>` at the end of this? It would mean that the birth and death messages would only color the beaver name, and the "was born" and "died of old age" part of the message is regular color.

Reply:
I think it's a great idea. Wasn't really sure how to achieve that.
I wonder if it will also not color the whole beaver any more.
Feel free to create a pull request with the change. Or I can do it in several hours, when I'm back from work.

Reply:
I think the character tint comes from `Timberborn.GameDistrictsUI.CitizenTint.UpdateTint` calling `Timberborn.CoreUI.ColorParser.TryGetColorFromText` with the beaver/bot/district name. The depends on the `#` being present, but I think named colors might also work, i.e. `<color=green>` will change the text color, but not the character tint (full list of named colors can be found at https://docs.unity3d.com/ScriptReference/ColorUtility.TryParseHtmlString.html)